src/uct/ib/ud/base/ud_iface.c

```diff
@@ -314,7 +356,8 @@ uct_ud_iface_conn_match_purge_cb(ucs_conn_match_ctx_t *conn_match_ctx,
     uct_ud_ep_t *ep       = ucs_container_of(elem, uct_ud_ep_t,
                                              conn_match);
 
-    ep->flags &= ~UCT_UD_EP_FLAG_ON_CEP;
+    ucs_assert(ep->flags & UCT_UD_EP_FLAG_ON_CEP);
+    ep->flags &= ~(UCT_UD_EP_FLAG_ON_CEP | UCT_UD_EP_FLAG_TX_RX);
```

Comment:
why need to remove TXRX?

Reply:
to not remove EP twice.
added the comment